### ISSUE TITLE
Provide key bindings for the numpad keys

### ISSUE THREAD

**Issue by ornamentist** [zed-industries/zed#11948]
### Check for existing issues

- [X] Completed

### Describe the feature

Could we have separate key bindings for the numpad keys? For example, `numpad_7`, `numpad_plus`, `numpad_enter`. My understanding is that they generate distinct keycodes on supported platforms.

I've gotten very used to binding actions/macros etc to a single key press on the numpad in other editors.

**Comment by ornamentist** [comment]
This is for macOS.

**Comment by elpixo** [comment]
And Linux too!

**Comment by phromo** [comment]
I want my numpad to become my AI suggestion manipulation pad!

**Comment by pkoppstein** [comment]
It's been more than 6 months ...

**Comment by coopbri** [comment]
> It's been more than 6 months ...

They are building a complex software with thousands of feature requests. Your comment doesn't benefit the discussion, be patient.

**Comment by pkoppstein** [comment]
@coopbri - My comment was primarily intended to help the developers prioritize the enhancement requests, but also to ensure that whatever bots rule this roost don't automatically close this "issue" as "stale".

**Comment by eugenesvk** [comment]
Another use case: since Zed is great in allowing binding anything to vim counts, you can bind regular inconvenient top number row to some rarely used commands and bind numpad numbers to `vim::Number` count commands (and add numpad numbers to you main alpha keyboard as an ergonomic block layer
uio
jkl
m,.

**Comment by olejorgenb** [comment]
Noting down some minimal investigations:

https://github.com/zed-industries/zed/blob/e4c90be58a44ff05152ac92a7bf35c67f05dafab/crates/gpui/src/platform/linux/platform.rs#L883

Reverting this substitution would presumably allow binding `kp_*` on linux. But this break using the numpad keys as normal keys. Eg. a `ctrl-1` binding would no longer work when using the keypad 1. (#12117)

It need to be either (1) a system which know that `kp_1` is equivalent to `1`, yet a distinct key. _If_ there is a `kp_*` binding it should take precedence. Otherwise the key should behave as the equivalent. 

or

(2) require that all `1` etc. bindings also have a `kp_1` binding. (https://github.com/zed-industries/zed/pull/13396)

(2) would be the simplest solution, but I assume it would be too confusing. Eg.: if users rebind a regular key (eg.: `ctrl-enter`) the numpad key equivalent would still be bound to the default action.

---

This [commit](https://github.com/zed-industries/zed/commit/0f7600e5a26096f86a911944f26801442fa7d21b) contains a POC for (1). It allows binding both `1` and `kp_1` **if** the `kp_1` binding come last:

```
  {
    "context": "Editor",
    "bindings": {
      "ctrl-1": "editor::ToggleLineNumbers"
    }
  },
  {
    "context": "Editor",
      "bindings": {
      "ctrl-numpad_1": "editor::ToggleRelativeLineNumbers"
    }
  }
```

It is not complete:  eg.: the "Search by keystroke" function does not work (though *binding* a numpad key in the keymap editor does work)  (UPDATE: it does work when using [exact matching](https://github.com/olejorgenb/zed/blob/0f7600e5a26096f86a911944f26801442fa7d21b/crates/keymap_editor/src/keymap_editor.rs#L598))

**Comment by eugenesvk** [comment]
what about a simpler version of (1)b: have a global `numlock_on` config to treat numpad keys as regular keys or separate? And duplicating the default keymaps with numpadX for every X? This would avoid confusion and only affect people who do rebind many keys (then they'd have to do their own duplication if they want to do so numpad bindings separately)

(by the way, Sublime Text separates the numpad keys separately as `keypadX`)

**Comment by olejorgenb** [comment]
> by the way, Sublime Text separates the numpad keys separately as keypadX

Ie.: They are treated as fully distinct keys?

I checked VSCode (or rather - Cursor) and it also seems to simply threat the numpad keys as distinct (1). (Except enter)

**Comment by eugenesvk** [comment]
Yes, fully distinct, so these bindings don't conflict, and without either of these the corresponding key doesn't work (so 1 isn't mirrored to also work with keypad1)
```json
{"keys":["ctrl+1"      ],"command":"move_to","args":{"to":"bol","extend" :false}},
{"keys":["ctrl+keypad1"],"command":"move_to","args":{"to":"eol","extend" :false}},
```
same with enter
```json
{"keys":["ctrl+enter"],"command":"move_to","args":{"to":"eol","extend" :false}},
{"keys":["ctrl+keypad_enter"],"command":"move_to","args":{"to":"bol","extend" :false}},
```
